service/history/command_checker.go

```diff
@@ -297,46 +298,48 @@ func (v *commandAttrValidator) validateActivityScheduleAttributes(
 	const failedCause = enumspb.WORKFLOW_TASK_FAILED_CAUSE_BAD_SCHEDULE_ACTIVITY_ATTRIBUTES
 
 	if attributes == nil {
-		return failedCause, serviceerror.NewInvalidArgument("ScheduleActivityTaskCommandAttributes is not set on command.")
+		return failedCause, serviceerror.NewInvalidArgument("ScheduleActivityTaskCommandAttributes is not set on ScheduleActivityTaskCommand.")
+	}
+
+	activityID := attributes.GetActivityId()
+	activityType := ""
+	if attributes.ActivityType != nil {
+		activityType = attributes.ActivityType.GetName()
 	}
 
 	defaultTaskQueueName := ""
 	if _, err := v.validateTaskQueue(attributes.TaskQueue, defaultTaskQueueName); err != nil {
-		return failedCause, err
+		return failedCause, errors.WithMessagef(err, "Invalid TaskQueue on ScheduleActivityTaskCommand: %v. ActivityId=%s ActivityType=%s", err, activityID, activityType)
```

Comment:
why not use `fmt.Errorf` with the `%w` specifier?

Reply:
No reason 🙂